### PR TITLE
Query UTXOs in testnet using ledger state

### PR DESCRIPTION
**Pull request by carbolymer**
# Description

Update cardano-api-8.39.0 & cardano-cli-8.20.2.0.

Use `foldEpochState` to provide live view on UTxO state in the testnet. This allows faster querying of UTxOs rather than going through CLI.

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] New tests are added if needed and existing tests are updated.  These may include:
  - golden tests
  - property tests
  - roundtrip tests
  - integration tests
  See [Runnings tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [ ] Any changes are noted in the `CHANGELOG.md` for affected package
- [ ] The version bounds in `.cabal` files are updated
- [ ] CI passes. See note on CI.  The following CI checks are required:
  - [ ] Code is linted with `hlint`.  See `.github/workflows/check-hlint.yml` to get the `hlint` version
  - [ ] Code is formatted with `stylish-haskell`.  See `.github/workflows/stylish-haskell.yml` to get the `stylish-haskell` version
  - [ ] Code builds on Linux, MacOS and Windows for `ghc-8.10.7` and `ghc-9.2.7`
- [ ] Self-reviewed the diff

# Note on CI
If your PR is from a fork, the necessary CI jobs won't trigger automatically for security reasons.
You will need to get someone with write privileges.  Please contact IOG node developers to do this
for you.

### CODE REVIEW

File: cardano-testnet/src/Testnet/Property/Utils.hs [hunk 1]
```diff
@@ -1,17 +1,25 @@
...
 
   , QueryTipOutput(..)
   , queryTip
-  , waitUntilEpoch
   , queryUtxos
+  , waitUntilEpoch
+  , getEpochStateView
+  , findAllUtxos
+  , findUtxosWithAddress
+  , findLargestUtxoWithAddress
```
Comment (smelc) [inline cardano-testnet/src/Testnet/Property/Utils.hs:22]: I think all these functions would be better placed in a sibling `Query.hs` module. Those functions are all used in the middle of tests (in the business logic) to obtain some data. I think that being in the `Utils.hs` file doesn't indicate how central those functions are to most integration tests.

Comment (carbolymer) [reply cardano-testnet/src/Testnet/Property/Utils.hs]: Moved to Query.hs

**Comment by NeilBurgess42** [comment]
I will clone the repo and push a commit with suggested changes.
One thing; IOG style is UTXO, not UTxO. The rationale is 'write it like you say it'. so UTXO is 'you tee ex oh', UTxO is 'you tix oh'.

**Review by NeilBurgess42 — APPROVED** [review]
The only suggested change is UTxO to UTXO in Readme.rst. Note, code files are out of scope for a technical writer; I can't be relied on to correctly differentiate between comments and operation code. I notice, though, that a user is going to see a lot of UTxO. In all other IOG documentation, users see UTXO and EUTXO.

**Review by Jimbo4350 — APPROVED** [review]
Nice work :+1:

One comment that needs addressing: https://github.com/IntersectMBO/cardano-node/pull/5662#discussion_r1502929252

File: cardano-testnet/src/Testnet/Components/Query.hs [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+  H.noteShowM $ H.jsonErrorFail $ fromJSON @QueryTipLocalStateOutput tipJSON
+
+newtype QueryTipOutput = QueryTipOutput { unQueryTipOutput :: FilePath }
+
+-- | A read-only mutable pointer to an epoch state, updated automatically
+newtype EpochStateView = EpochStateView (IORef AnyNewEpochState)
+
+-- | Create a background thread listening for new epoch states. New epoch states are available to access
+-- through 'EpochStateView', using query functions.
+getEpochStateView
```
[the remark this answers is not in the document]
Comment (carbolymer) [reply cardano-testnet/src/Testnet/Components/Query.hs:104]: I thought it would be a type easier to understand. But yeah `Maybe` could work here better.

**Review by Jimbo4350 — APPROVED** [review]

**Comment by carbolymer** [comment]
@NeilBurgess42 thanks! I followed ledger's naming convention, where [they're using `UTxO` in the codebase](https://github.com/search?q=repo%3AIntersectMBO%2Fcardano-ledger+UTXO+language%3AHaskell&type=code&l=Haskell). I think we should switch to all-capitalised UTXO throughout our codebase at some point in a piecemeal fashion. I've updated my commit and PR names to use "UTXO".

**Pull request merged**